mysql-test/suite/galera/t/galera_restart_replica.test

```diff
@@ -0,0 +1,138 @@
+--source include/big_test.inc
+--source include/force_restart.inc
+--source include/galera_cluster.inc
+--source include/have_sequence.inc
+
+#
+
+#
+# We use two 2-node galera clusters as follows
+#
+#       node_1 <-> node_2              {Galera cluster 1}
+#                                      {Async replication}
+#       node_3 <-> node_4              {Galera cluster 2}
+#
+# Normal asyncronous replication is used between nodes 1 and 4
+# so that node_1 is master and node_4 a slave.
+#
+#
+--connect node_3, 127.0.0.1, root, , test, $NODE_MYPORT_3
+--connect node_4, 127.0.0.1, root, , test, $NODE_MYPORT_4
+
+--let $node_1 = node_1
+--let $node_2 = node_2
+--let $node_3 = node_3
+--let $node_4 = node_4
+--source include/auto_increment_offset_save.inc
+
+--connection node_2
+call mtr.add_suppression("WSREP: Ignoring server id for non bootstrap node.");
+--connection node_4
+call mtr.add_suppression("WSREP: Ignoring server id for non bootstrap node.");
+
+--connection node_4
+--replace_result $NODE_MYPORT_1 NODE_MYPORT_1
+--eval CHANGE MASTER TO master_host='127.0.0.1', master_user='root', master_port=$NODE_MYPORT_1, master_use_gtid=current_pos;
+START SLAVE;
+--source include/wait_for_slave_to_start.inc
+
+--connection node_1
+
+CREATE TABLE t1 (id bigint primary key, msg varchar(100)) engine=innodb;
+--disable_query_log
+INSERT INTO t1 SELECT seq, 'test' from seq_1_to_50000;
+--enable_query_log
+SELECT COUNT(*) FROM t1;
+--source include/save_master_gtid.inc
+
+--connection node_3
+# Set higher weight for node_3 to keep it in primary
+# while node_4 is dead
+SET GLOBAL wsrep_provider_options = 'pc.ignore_sb=true';
+SET GLOBAL wsrep_provider_options = 'pc.weight=2';
+
+--echo # Waiting table and rows to replicate to node_3
+--let $wait_condition = SELECT COUNT(*) = 1 FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_NAME = 't1'
+--source include/wait_condition.inc
+--let $wait_condition = SELECT COUNT(*) = 50000 FROM t1
+--source include/wait_condition.inc
+SELECT COUNT(*) FROM t1;
+
+--connection node_4
+--source include/sync_with_master_gtid.inc
+--echo # Waiting table and rows to replicate to node_4
+--let $wait_condition = SELECT COUNT(*) = 1 FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_NAME = 't1'
+--source include/wait_condition.inc
+--let $wait_condition = SELECT COUNT(*) = 50000 FROM t1
+--source include/wait_condition.inc
+SELECT COUNT(*) FROM t1;
+
+--connection node_4
+# Isolate node_4 from the group and wait until wsrep_ready becomes OFF.
+SET GLOBAL wsrep_provider_options = 'gmcast.isolate = 1';
```

Comment:
using gmcast.isolate looks obsolete here, the 2 node cluster should be fine to shrink to 1 node cluster if node_2 is just stopped